src/model/relationship/RelationshipImpl.java

```diff
@@ -0,0 +1,190 @@
+package model.relationship;
+
+import model.GeneologyRules;
+import model.person.IPerson;
+
+import java.util.Random;
+
+public class RelationshipImpl implements IRelationship {
+  private double regard;
+  private double desire;
+  private IPerson p1;
+  private IPerson p2;
+  private int anniversaryYear;
+  private RelationshipType romanticType;
+
+  public RelationshipImpl(IPerson p1, IPerson p2, int anniversaryYear) {
+    this(p1, p2, GeneologyRules.computeRegard(p1, p2), GeneologyRules.computeDesire(p1, p2), anniversaryYear, null);
+  }
+
+  public RelationshipImpl(IPerson p1, IPerson p2, double regard, double desire, int anniversaryYear, RelationshipType romanticType) {
+    this.regard = regard;
+    this.desire = desire;
+    this.p1 = p1;
+    this.p2 = p2;
+    this.anniversaryYear = anniversaryYear;
+    this.romanticType = romanticType;
+  }
+
+
+  @Override
+  public IPerson p1() {
+    return this.p1;
+  }
+
+  @Override
+  public IPerson p2() {
+    return this.p2;
+  }
+
+  @Override
+  public double regard() {
+    return this.regard;
+  }
+
+  @Override
+  public void growRegard() {
+    this.regard = Math.min(1, this.regard + 0.1);
+  }
+
+  @Override
+  public void lessenRegard() {
+    this.regard = Math.max(-1, this.regard - 0.1);
+  }
+
+  @Override
+  public double romanticDesire() {
+    return this.desire;
+  }
+
+  @Override
+  public void growDesire() {
+    this.desire = Math.min(1, this.desire + 0.1);
```

Comment:
Maybe make 0.1 a constant so we can mess around with it more easily later.

Reply:
yep, definitely can do that